taskiq_redis/redis_broker.py

```diff
@@ -82,25 +82,24 @@ async def listen(
 
         :param callback: function to call on new message.
         """
+        REDIS_BRPOP_DATA_POSITION = 1
         loop = asyncio.get_event_loop()
         async with Redis(connection_pool=self.connection_pool) as redis_conn:
-            redis_pubsub_channel = redis_conn.pubsub()
-            await redis_pubsub_channel.subscribe(self.redis_pubsub_channel)
-            async for message in redis_pubsub_channel.listen():
-                if message:
-                    try:
-                        redis_message = pickle.loads(
-                            message["data"],
-                        )
-                        if isinstance(redis_message, BrokerMessage):
-                            loop.create_task(callback(redis_message))
-                    except (
-                        TypeError,
-                        AttributeError,
-                        pickle.UnpicklingError,
-                    ) as exc:
-                        logger.debug(
-                            "Cannot read broker message %s",
-                            exc,
-                            exc_info=True,
-                        )
+            while True:
+                message = (await redis_conn.brpop(self.queue_name))[
+                    REDIS_BRPOP_DATA_POSITION
+                ]
+                try:
+                    redis_message = pickle.loads(message)
+                    if isinstance(redis_message, BrokerMessage):
+                        loop.create_task(callback(redis_message))
```

Comment:
I'm not sure, but there can be a potential bug  
Due to official documentation - https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task (take a look at **Important**), if we create a task and don't save it to any sequence to create a new reference to the task, the task can be dropped by the garbage collector before execution.

Before you pull the request we used the same way but I think it's better to fix now.

Reply:
I was just writing about it)
https://github.com/taskiq-python/taskiq-redis/issues/13

Reply:
I guess we need to update API for brokers.

Reply:
Maybe just:
`await callback(redis_message)`
?

Reply:
In this case, we will lose the full benefit of asynchrony, I think it's better to add a new set as mentioned in the doc, and add a callback to tasks.

Reply:
made the changes as in the example